src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -284,7 +284,6 @@ private Object createObjectFromFullModuleNames(String name, Set<String> fullModu
         }
         else {
             final String optionalNames = fullModuleNames.stream()
-                    .sorted()
```

Comment:
this simply means we need to extend valiadtion of excpetion message to be check that modules are sorted.

Reply:
here is test for this:
https://github.com/checkstyle/checkstyle/blob/0b3fdfa7a5b574d73bd4e46b76112946235c48b6/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java#L239-L260

we need to extend assert to make sure name are ordered.

I found this method by simply placing breakpoint for this code and execution of PackageObjectFactoryTest tests.

Reply:
Lets suppose https://github.com/checkstyle/checkstyle/pull/13927#discussion_r1367911022 for the test case if we change the optionalValue String in unsorted way and we remove the sorted function the test case will still fail 

I have gone this deep into this method the result value is already sorted. I am looking into it how this is happening.
https://github.com/checkstyle/checkstyle/blob/eacee5871c1a58a7bfce8033c081696c3240f89b/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java#L305C1-L318

Reply:
make code like:
```
✔ ~/java/github/romani/checkstyle [Kevin222004/PackageObjectFactory2 L|✚ 2…1⚑ 1] 
19:14 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java b/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java
index 314b1a2..a2ecdeb 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java
@@ -285,6 +285,7 @@ public class PackageObjectFactory implements ModuleFactory {
         else {
             final String optionalNames = fullModuleNames.stream()
                     .collect(Collectors.joining(STRING_SEPARATOR));
+            System.out.println("fullModuleNames: " + fullModuleNames);
             final LocalizedMessage exceptionMessage = new LocalizedMessage(
                     Definitions.CHECKSTYLE_BUNDLE, getClass(),
                     AMBIGUOUS_MODULE_NAME_EXCEPTION_MESSAGE, name, optionalNames);
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java
index fa45940..e8e2a4d 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java
@@ -239,9 +239,11 @@ public class PackageObjectFactoryTest {
     public void testCreateObjectFromFullModuleNamesWithAmbiguousException() {
         final String barPackage = BASE_PACKAGE + ".packageobjectfactory.bar";
         final String fooPackage = BASE_PACKAGE + ".packageobjectfactory.foo";
+        final String fooPackage2 = BASE_PACKAGE + ".packageobjectfactory.zoo";
+        final String fooPackage3 = BASE_PACKAGE + ".packageobjectfactory.abc";
         final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
         final PackageObjectFactory objectFactory = new PackageObjectFactory(
-                new LinkedHashSet<>(Arrays.asList(barPackage, fooPackage)), classLoader);
+                new HashSet<>(Arrays.asList(fooPackage, barPackage, fooPackage2, fooPackage3)), classLoader);
         final String name = "FooCheck";
         try {
             objectFactory.createModule(name);

```

and create  FooCheck.java in each of such crazy package.

Reply:
done